addons/mail/static/src/less/thread.less

```diff
@@ -69,23 +69,24 @@
                 opacity: 0;
             }
         }
-
-
-        i.o_thread_message_star, i.o_thread_message_needaction, i.o_thread_message_reply, i.o_thread_message_email  {
+        .o_thread_icons {
             cursor: pointer;
-            opacity: 0;
-            &.fa-star {
+            .fa-star {
                 opacity: @mail-thread-icon-opacity;
                 color: gold;
             }
+            :not(.fa-star) {
```

Comment:
Rules which end with an alone `:not` are bad (ok that's better than four rules like before anyway but there is better than this too). I don't know the structure but something like this should be done:
```less
.o_thread_icons {
    cursor: pointer;
    > .o_thread_icon {
        ...
        &.fa-star {
            ...
        }
    }
}
```